src/signal.rs

```diff
@@ -129,32 +105,26 @@ fn SIGRTMAX() -> c_int {
 
 /// Verify that a signal number is valid.
 ///
-/// VCPU signals must be enclosed within the OS
-/// limits for realtime signals. The remaining ones need
-/// to be between the minimum (`SIGHUP`) and maximum (`SIGSYS`) values.
+/// Supported signals range from `SIGHUP` to `SIGUSR2` and from `SIGRTMIN` to `SIGRTMAX`.
+/// We recommend using realtime signals `[SIGRTMIN, SIGRTMAX]` for VCPU threads.
 ///
 /// # Arguments
 ///
 /// * `num`: the signal number to be verify.
-/// * `for_vcpu`: if `num` is a vcpu signal.
 ///
 /// # Examples
 ///
 /// ```
 /// extern crate vmm_sys_util;
 /// use vmm_sys_util::signal::validate_signal_num;
 ///
-/// let num = validate_signal_num(1, true).unwrap();
+/// let num = validate_signal_num(1).unwrap();
 /// ```
-pub fn validate_signal_num(num: c_int, for_vcpu: bool) -> errno::Result<c_int> {
-    if for_vcpu {
-        let actual_num = num + SIGRTMIN();
-        if actual_num <= SIGRTMAX() {
-            return Ok(actual_num);
-        }
-    } else if SIGHUP <= num && num <= SIGSYS {
+pub fn validate_signal_num(num: c_int) -> errno::Result<c_int> {
```

Comment:
Sorry for the late feedback, but since this function is now just validating and not mutating the signal number, how about we change it to returning a `bool` or a `Result<()>` (I prefer the latter):

```suggestion
pub fn validate_signal_num(num: c_int) -> errno::Result<()> {
```

Reply:
Yep, sure. I chose to return `Result<()>` because it is compatible with unit tests, and could be promptly returned in case of `Error`, while returning `bool` needs writing checks and explicit return of Error if the function call returns `false`.